rust_src/src/chartable.rs

```diff
@@ -255,36 +279,39 @@ impl LispSubCharTableRef {
 
         val
     }
+}
 
-    pub fn equal(self, other: Self, kind: equal_kind::Type, depth: i32, ht: LispObject) -> bool {
-        unsafe {
-            let mut size1 =
-                self.header.size as usize & More_Lisp_Bits::PSEUDOVECTOR_SIZE_MASK as usize;
-            let size2 =
-                other.header.size as usize & More_Lisp_Bits::PSEUDOVECTOR_SIZE_MASK as usize;
-            if size1 != size2 {
-                return false;
-            }
-
-            size1 -= 2; // account for depth and min_char
-            if self.depth != other.depth {
-                return false;
-            }
-            if self.min_char != other.min_char {
-                return false;
-            }
+impl LispStructuralEqual for LispSubCharTableRef {
+    fn equal(
+        &self,
+        other: Self,
+        kind: equal_kind::Type,
+        depth: i32,
+        ht: &mut LispHashTableRef,
+    ) -> bool {
+        let mut size1 =
+            unsafe { self.header.size as usize & More_Lisp_Bits::PSEUDOVECTOR_SIZE_MASK as usize };
+        let size2 =
+            unsafe { other.header.size as usize & More_Lisp_Bits::PSEUDOVECTOR_SIZE_MASK as usize };
+        if size1 != size2 {
+            return false;
+        }
 
-            let slice1 = self.contents.as_slice(size1);
-            let slice2 = other.contents.as_slice(size1);
-            for i in 0..size1 {
-                let v1 = slice1[i];
-                let v2 = slice2[i];
-                if !internal_equal(v1, v2, kind, depth + 1, ht) {
-                    return false;
-                }
-            }
+        size1 -= 2; // account for depth and min_char
+        if self.depth != other.depth {
+            return false;
+        }
+        if self.min_char != other.min_char {
+            return false;
         }
-        true
+
+        let slice1 = unsafe { self.contents.as_slice(size1) };
+        let slice2 = unsafe { other.contents.as_slice(size1) };
```

Comment:
At some point, someone will need to make a method on LispSubCharTableRef for getting the contents as a slice.